libs/fluxc-annotations/build.gradle

```diff
@@ -1,26 +1,8 @@
 plugins {
     id "java"
```

Comment:
Thanks, but doesn't it feel redundant in this case? We write more code in 2 files instead of simply having `id "java"` in one 🤔 What are the benefits to use catalog and alias in this case?

Reply:
Yea, I understand what you are saying, but for me it is about:
1. Consistency -> Having `alias(libs.plugins.xzy)` everywhere and having everyone adhering to this standard, that is, instead of a random `id "xyz"`. Think constants definition but for builds.
2. Updatability -> Keep that `id` string in one place and one place only, version catalogs for that matter, then, when and if needed, update it in that one place, that is, instead of multiple places.

Reply:
I see, thanks for sharing, but frankly speaking this doesn't convince me:

> Consistency -> Having alias(libs.plugins.xzy) everywhere and having everyone adhering to this standard, that is, instead of a random id "xyz". Think constants definition but for builds.

I don't think `id "java"` is random - it's a standard way of declaring plugins in Gradle. It's already a constant in this sense.

> Updatability -> Keep that id string in one place and one place only, version catalogs for that matter, then, when and if needed, update it in that one place, that is, instead of multiple places.

I can't imagine that `java` plugin will change its name. If this, highly unlikely IMO, scenario happen, we'll have to only update 2 files, so I think we shouldn't worry about it.

Taking both into account, for now I'd not introduce java to version catalog, as I simply don't see benefits and I see downsides: boilerplate code.

Reply:
Totally, up to you, but honestly, for example, if I where to start a new project, I would put `java` into version catalogs and not have that there be different from the rest.

1. About the random thing, I was trying to make an analogy with constants. We've already used to move any magic number or random strings into a constant and use that constant instead, I think there is no debate about it, is it? No matter if it is only one string within a class file, the recommendation would be move that to a constant (static analysis, etc). Same with this `java` plugin. To me, it is a plain constant that could be moved into version catalogs.
2. About the `java` plugin changing, think about the `kotlin` id, and the fact that it did indeed change, for example, `org.jetbrains.kotlin.jvm` is becoming the recommending id to use.

Now, my main struggle here is what do we do when I made that change, moving `java` into version catalogs, in some other project, would you come at me to move it back to where it was? 🤔 That's why I am trying here to align on it, and any other minor suggestion such as this, I don't mind what will happen here and now, but rather next and if the situation reverses.

PS: Thanks for the discussion, as always, but please ignore it, you don't have to act on it if you don't feel like you need to, I totally understand.